src/com/amazon/ion/impl/bin/utf8/Pool.java

```diff
@@ -22,14 +24,19 @@
     private static final int MAX_QUEUE_SIZE = 128;
 
     // A queue of previously initialized objects that can be loaned out.
-    private final ArrayBlockingQueue<T> bufferQueue;
+    private final Queue<T> bufferQueue;
```

Comment:
I notice some pre-existing comments/variables mention things like `buffer` and `block`; these aren't accurate for the generalized `Pool<T>` type.

```suggestion
    private final Queue<T> objectQueue;
```